packages/devtools_app/lib/src/screens/memory/panes/chart/chart_pane.dart

```diff
@@ -0,0 +1,706 @@
+          final timestamp = tapData.timestamp!;
+
+          final copied = TapLocation.copy(tapLocation);
+          widget.chartControllers.event.tapLocation.value = copied;
+          widget.chartControllers.vm.tapLocation.value = copied;
+
+          final allValues = ChartsValues(controller, index, timestamp);
+
+          _showHover(
+            context,
+            allValues,
+            tapData.tapDownDetails!.globalPosition,
+          );
+        }
+      }
+    });
+
+    addAutoDisposeListener(controller.refreshCharts, () {
+      setState(() {
+        _refreshCharts();
+      });
+    });
+
+    addAutoDisposeListener(controller.androidChartVisibleNotifier, () {
+      setState(() {});
+    });
+
+    // Update the chart when the memorySource changes.
+    addAutoDisposeListener(controller.memorySourceNotifier, () async {
+      try {
+        await controller.updatedMemorySource();
+      } catch (e) {
+        final errorMessage = '$e';
+        controller.memorySource = MemoryController.liveFeed;
+        // Display toast, unable to load the saved memory JSON payload.
+        final notificationsState = Notifications.of(context);
+        if (notificationsState != null) {
+          notificationsState.push(errorMessage);
+        } else {
+          // Running in test harness, unexpected error.
+          throw OfflineFileException(errorMessage);
+        }
+        return;
+      }
+
+      controller.refreshAllCharts();
+    });
+
+    _updateListeningState();
+  }
+
+  void _updateListeningState() async {
+    await serviceManager.onServiceAvailable;
+
+    if (!controller.hasStarted) {
+      await controller.startTimeline();
+
+      // TODO(terry): Need to set the initial state of buttons.
+/*
+      pauseButton.disabled = false;
+      resumeButton.disabled = true;
+
+      vmMemorySnapshotButton.disabled = false;
+      resetAccumulatorsButton.disabled = false;
+      gcNowButton.disabled = false;
+
+      memoryChart.disabled = false;
+*/
+    }
+  }
+
+  void _refreshCharts() {
+    // Remove history of all plotted data in all charts.
+    widget.chartControllers.event.reset();
+    widget.chartControllers.vm.reset();
+    widget.chartControllers.android.reset();
+
+    _recomputeChartData();
+  }
+
+  /// Recompute (attach data to the chart) for either live or offline data source.
+  void _recomputeChartData() {
+    widget.chartControllers.event.setupData();
+    widget.chartControllers.event.dirty = true;
+    widget.chartControllers.vm.setupData();
+    widget.chartControllers.vm.dirty = true;
+    widget.chartControllers.android.setupData();
+    widget.chartControllers.android.dirty = true;
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Column(
+      children: [
+        const SizedBox(height: denseRowSpacing),
+        SizedBox(
+          height: scaleByFontFactor(70),
+          child: events.MemoryEventsPane(widget.chartControllers.event),
+        ),
+        SizedBox(
```

Comment:
Is this SizedBox even necessary? I suspect it isn't, but if it is, please move into MemoryVMChart.build directly